packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -501,25 +478,24 @@ class NavigatorState extends State<Navigator> {
     return _history.length > 1 || _history[0].willHandlePopInternally;
   }
 
-  bool _hadTransaction = true;
+  final Set<int> _activePointers = new Set<int>();
 
-  /// Calls callback immediately to create a navigator transaction.
-  ///
-  /// To avoid race conditions, a navigator will execute at most one operation
-  /// per animation frame. If you wish to perform a compound change to the
-  /// navigator's state, you can use a navigator transaction to execute all the
-  /// changes atomically by making the changes inside the given callback.
-  bool openTransaction(NavigatorTransactionCallback callback) {
-    assert(callback != null);
-    if (_hadTransaction)
-      return false;
-    _hadTransaction = true;
-    NavigatorTransaction transaction = new NavigatorTransaction._(this);
-    setState(() {
-      callback(transaction);
-    });
-    assert(() { transaction._debugClose(); return true; });
-    return true;
+  void _handlePointerDown(PointerDownEvent event) {
+    _activePointers.add(event.pointer);
+  }
+
+  void _handlePointerUpOrCancel(PointerEvent event) {
+    _activePointers.remove(event.pointer);
+  }
+
+  void _cancelActivePointers() {
+    BuildContext overlayContext = _overlayKey.currentContext;
+    if (overlayContext != null) {
+      RenderAbsorbPointer absorber = overlayContext.ancestorRenderObjectOfType(const TypeMatcher<RenderAbsorbPointer>());
+      absorber?.absorbing = true;
+    }
+    for (int pointer in _activePointers.toList())
+      WidgetsBinding.instance.cancelPointer(pointer);
```

Comment:
This will also cancel gesture recognisers dealing with those pointers outside the navigator.